src/app/shared/components/workshop-checkbox-dropdown/workshop-checkbox-dropdown.component.html

```diff
@@ -1,14 +1,14 @@
 <mat-form-field class="dropdown-wrapper">
-  <mat-label class="label" [ngClass]="{'disabled-field': workshops?.length < 1}">Усі гуртки</mat-label>
-  <mat-select [disabled]="workshops?.length < 1" [formControl]="workshopControl" multiple class="dropdown" disableOptionCentering
-    panelClass="dropdown-panel application-dropdown {{ dropdownContainerClass }}">
+  <mat-label class="label" [ngClass]="{'disabled-field': entities?.length < 1}"> Усі {{entities?.length | declination: Declination }}</mat-label>
+  <mat-select [disabled]="entities?.length < 1" [formControl]="entityControl" multiple class="dropdown" disableOptionCentering
+    panelClass="dropdown-panel application-dropdown ">
     <mat-select-trigger>
-      <div *ngIf="workshopControl.value?.length">
-        <span *ngIf="workshopControl.value?.length > 1"> {{workshopControl.value?.length | declination: workshopDeclination}} вибрано </span>
-        <span *ngIf="workshopControl.value?.length === 1"> {{workshopControl.value[0].title}} </span>
+      <div *ngIf="entityControl.value?.length">
+        <span *ngIf="entityControl.value?.length > 1"> {{entityControl.value?.length | declination: Declination}} вибрано </span>
+        <span *ngIf="entityControl.value?.length === 1"> {{entityControl.value[0].title || entityControl.value[0].firstName}} </span>
       </div>
     </mat-select-trigger>
-    <mat-option class="dropdown-option" *ngFor="let workshop of workshops" [value]="workshop">{{ workshop.title }}
+    <mat-option class="dropdown-option" *ngFor="let entity of entities" [value]="entity">{{ entity.firstName ? getFullName(entity.firstName, entity.lastName) : entity.title }}
```

Comment:
if you are already calling a method from the template, then move the entity.firstName check to this method, to the template, won't have any additional logic
you can check what you need to return (title or full name) in the method